Switch to conda-build, Add dep information

milljm opened the pull request:
Switch from `mamba mambabuild` to `conda-build` in the generate script. We will someday do the same for Civet recipes...

Add dependency information to empty_shell template. This will be necessary for an upcoming change to ncrc-client. This increases the package size by a minimal amount (all the meta data about what this package requires, will now be in this package). The time it takes to build this package also seems to be unaffected.

Refs #26343

Commit:
Switch to conda-build, Add dep information

Switch from `mamba mambabuild` to `conda-build` in the generate script.
We will someday do the same for Civet recipes...

Add dependency information to empty_shell template. This will be
necessary for an upcoming change to ncrc-client. This increases the
package size by a minimal amount (all the meta data about what this
package requires, will now be in this package). The time it takes
to build this package also seems to be unaffected.

Refs #26343

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1951430/) on bb970ad wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/26362/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1951445/) on bb970ad wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>3b9776</th>
      <th colspan=4>#26362 bb970a</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.06%</td>
      <td>85.06%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>98122</td>
      <td>98121</td>
      <td>-1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>17232</td>
      <td>17233</td>
      <td>+1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26362/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26362/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/26362/coverage/xfem)

</details>



This comment will be updated on new commits.

cticenhour commented on conda/template/conda_build_config.yaml:
```diff
@@ -1,19 +0,0 @@
-moose_dev:
```
With this file being turned into a symlink, the header information in the relevant `meta.yaml` files that mention modifying both the `moose/conda_build_config.yaml` and this one should be adjusted.

Commit:
Remove hint instruction

Remove the bit about modifying a now symlinked file. The only file
being mentioned is all that needs to be modified.

Closes #26343

milljm commented on conda/empty_shell/conda_build_config.yaml:
```diff
@@ -0,0 +1 @@
+../template/conda_build_config.yaml
```
I debated whether not to link to the actual file or the one that might really matter if it changes...
